### DISCUSSION TITLE
A New Feature to Account for UR in the Judgement System

### DISCUSSION THREAD

**Discussion by Rekunan** [ppy/osu#17283] [Ideas]
(I can't recall if Etterna's wife system has this feature, as I vaguely recall watching a vid that mentioned this a while ago, which led me to want this to be in osu)

If a note was hit within the 300 window, but then the next set of notes were all hit consistently (let's say early for this example) within the 100 window, then only the first note in that set would give a 100 with the rest giving 300s. Then, let's say that the following set of notes were all hit within the original 300 window, then again, only the first note of that set would give a 100 with the rest giving 300s.

To give an example, let's say that I start a 16-note stream early, and since I don't want to mess up my stamina or tapping by offsetting my tapping, I consistently hit all the notes in that stream early, which would then become a full stream of 100s. But if this gets implemented, only note 1 of that stream would give a 100, and notes 2-16 would give 300s. Then, let's say that there's a 4-note set of jumps after the stream that I hit on time within the original 300 window, then note 1 in that set of jumps would give a 100, but notes 2-4 would give 300s.

**Comment by bdach** [comment]
This seems like a weird mechanic to have. So you could theoretically get 300s for... not being on time, as long as you're _consistently_ early/late? Not only does that seem really bad on the surface (you're still *not on time* but not getting punished for it), but isn't that also opening up an avenue for cheesing/manipulation?

**Reply by Sublimelime** [reply]
> So you could theoretically get 300s for... not being on time, as long as you're consistently early/late? 

Yeah, from my understanding this change would _make timing relative to the previous object_, instead of relative to the song/overall timing of the beatmap.

I also don't think it's a good idea. Perhaps added as a mod, though adaptive speed does something similar to this (eg hitting too early makes the map speed up until it's no longer early). If added as a mod it could be called something like "Adaptive offset" or "Relative timing".

**Comment by peppy** [comment]
This is not something I would ever allow into osu! as it goes against the core ideals of a rhythm game (I would not even consider this as a mod). You're welcome to discuss further but I won't be following this thread.
